src/binary/binaryFetcher/binaryValidator.ts

```diff
@@ -0,0 +1,36 @@
+import * as fs from "fs";
+import { runProcess } from "../runProcess";
+// import { runProcess } from "../runProcess";
+
+export default async function isValidBinary(version: string): Promise<boolean> {
+  const exists = fs.existsSync(version);
+  if (!exists) {
+    return false;
+  }
+  const { proc, readLine } = runProcess(version, ["--print-version"]);
+
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      console.error(`validating ${version} timeout out`);
+      resolve(false);
```

Comment:
why use resolve with boolean over using reject?

Reply:
reject will break the execution with exception  until it caught, and here  i want to go over each directory and check if it is valid